libs/sqlserver/langchain_sqlserver/vectorstores.py

```diff
@@ -824,16 +849,36 @@ def add_texts(
             texts: Iterable of strings to add into the vectorstore.
             metadatas: List of metadatas (python dicts) associated with the input texts.
             ids: List of IDs for the input texts.
+            batch_size: Number of documents to be inserted at once to Db, max 419.
             **kwargs: vectorstore specific parameters.
 
         Returns:
             List of IDs generated from adding the texts into the vectorstore.
         """
-        # Embed the texts passed in.
-        embedded_texts = self.embedding_function.embed_documents(list(texts))
+        if texts is None:
+            return []
+
+        # Initialize a list to store results from each batch
+        embedded_texts = []
+
+        # Loop through the list of documents and process in batches
+        texts = list(texts)
+
+        # Validate batch_size again to confirm if it is still valid.
+        batch_size = self._validate_batch_size(self._batch_size)
+        for i in range(0, len(list(texts)), batch_size):
```

Comment:
isnt texts already a list(texts) as per 865? If yes, shouldnt this be 0, len(texts) ?